packages/plugin-legacy/README.md

```diff
@@ -122,6 +122,18 @@ export default {
   }
   ```
 
+### `fallbackToLegacyIfNoDynamicImport`
+
+- **Type:** `boolean`
+- **Default:** `false`
+
+  By default, plugin-legacy enables `polyfillDynamicImport` option, which injects lightweight dynamic import polyfill. If you want to disable this behavior and use native `import()`, but still support browsers with native ESM (e.g. legacy Edge) you can use this option.
+
+  Enabling this injects runtime check and loads legacy bundle with SystemJs runtime in browsers that have native ESM but no dynamic import support. This comes with the following drawbacks:
+
+  - Modern bundle is still downloaded in all ESM browsers
+  - Modern bundle throws `SyntaxError` in browsers without dynamic import
```

Comment:
Thanks for the work on this PR. We should try to think of a shorter name for the option, but it isn't easy. Maybe just `dynamicImportFallback`? @antfu any ideas here?

I think the option should be true by default, and the majority shouldn't think about this.

plugin-legacy should also stop enabling the polyfillDynamicImport option by default.

Reply:
I've renamed the option and made it true by default, although I'm unsure about this, since it introduces SyntaxErrors, which might be surprising.

Also I'm not sure what to do with `polyfillDynamicImport` option if `dynamicImportFallback` is disabled. Should we set it to true, leave false, or introduce a new option? I left the previous behaviour for now (setting it to true)

Reply:
These syntax errors appear in less than 0.5% of the browsers, and if we don't have this default there is a possibility that these browsers will be broken (probably without the user realizing it). So I think this is safer. The user can opt-out if needed.

You are right about `polyfillDynamicImport`, probably better to leave this logic and set it to true in case `dynamicImportFallback` is disabled. Another option is to have an option like:

```
dynamicImport: 'fallback' | 'polyfill' | 'none'
```

And deprecate the `polyfillDynamicImport` option in Vite. I think we can leave it like you proposed for the moment, and we can check with Evan before merging.

Reply:
@nulladdict we discussed with the team and we think that it's better to avoid introducing a new option (`polyfillDynamicImport`) until it is strictly required. It shouldn't be the case that people decide to explicitly not support 0.5% of the browsers if they are already choosing to use plugin-legacy. If the error is an issue, a global error handler could be added by the user, but even this shouldn't be a need.

So, the idea is to remove `polyfillDynamicImport` from the legacy config (we can leave the logic inside, just avoid exposing it). 

We also should remove all the logic related to enabling `polyfillDynamicImport` for plugin-legacy. In a future PR, we are also going to remove this feature from Core now that it is no longer needed. 

With these changes, I think we should be able to merge this PR. Thanks again for this work, and for the discussions that are helping to move this forward.

Reply:
I removed the option from the config, removed logic that enabled `polyfillDynamicImport` for the modern build, and made some adjustments to the readme.

I also rebased to the current main and squashed all the commits into one.

I think that's all the changes that were needed. Correct me if I missed something.